### ISSUE TITLE
StackOverflow in nextflow 22.04.1

### ISSUE THREAD

**Issue by scottgigante-immunai** [nextflow-io/nextflow#2872]
### Expected behavior and actual behavior

Expected: runs. Actual: StackOverflow.

### Steps to reproduce the problem

https://github.com/openproblems-bio/openproblems/runs/6442669625?check_suite_focus=true

### Program output 

[nextflow.log](https://github.com/nextflow-io/nextflow/files/8695743/nextflow.log)

### Environment 

* Nextflow version: 22.04.1
* Java version: 15
* Operating system: Ubuntu 18.04
* Bash version: v4.4.20(1)-release

### Additional context

Problem began 7 hours ago when v22.04.1 was released, worked fine before then. Forced install of 22.04.0 and now it works again.

**Comment by pditommaso** [comment]
Maybe related to #2815

**Comment by ewels** [comment]
Also seen with @nf-core / chipseq by @drpatelh:

```console
$ nextflow run nf-core/chipseq -profile test,docker
N E X T F L O W  ~  version 22.04.1
Pulling nf-core/chipseq ...
 downloaded from https://github.com/nf-core/chipseq.git
Unexpected error [StackOverflowError]
```

**Comment by pditommaso** [comment]
Does it happen with 22.04.0?

**Comment by pditommaso** [comment]
Oh a stack overflow in the Java regexp parser

```
java.lang.StackOverflowError: null
	at java.base/java.lang.Character.codePointAt(Character.java:8867)
	at java.base/java.util.regex.Pattern$CharProperty.match(Pattern.java:3992)
	at java.base/java.util.regex.Pattern$Branch.match(Pattern.java:4800)
	at java.base/java.util.regex.Pattern$GroupHead.match(Pattern.java:4855)
	at java.base/java.util.regex.Pattern$Loop.match(Pattern.java:4964)
	at java.base/java.util.regex.Pattern$GroupTail.match(Pattern.java:4886)
	at java.base/java.util.regex.Pattern$BranchConn.match(Pattern.java:4763)
	at java.base/java.util.regex.Pattern$CharProperty.match(Pattern.java:3996)
	at java.base/java.util.regex.Pattern$Branch.match(Pattern.java:4800)
	at java.base/java.util.regex.Pattern$GroupHead.match(Pattern.java:4855)
	at java.base/java.util.regex.Pattern$Loop.match(Pattern.java:4964)
	at java.base/java.util.regex.Pattern$GroupTail.match(Pattern.java:4886)
	at java.base/java.util.regex.Pattern$BranchConn.match(Pattern.java:4763)
	at java.base/java.util.regex.Pattern$CharProperty.match(Pattern.java:3996)
	at java.base/java.util.regex.Pattern$Branch.match(Pattern.java:4800)
	at java.base/java.util.regex.Pattern$GroupHead.match(Pattern.java:4855)
	at java.base/java.util.regex.Pattern$Loop.match(Pattern.java:4964)
	at java.base/java.util.regex.Pattern$GroupTail.match(Pattern.java:4886)
	at java.base/java.util.regex.Pattern$BranchConn.match(Pattern.java:4763)
```

![image](https://user-images.githubusercontent.com/816968/168644180-9458754a-c91e-4903-ad85-3f32a209e504.png)

**Comment by JoseEspinosa** [comment]
Nope, it does work with the following command: 
`NXF_VER=22.04.0 nextflow run nf-core/chipseq -profile test,docker -r 1.2.2 -dsl1`

**Comment by ewels** [comment]
I don't think so no - I'm getting an error because I need to update Java but it gets past the same point:

```console
$ NXF_VER=22.04.1 nextflow run nf-core/chipseq -profile test,docker
NOTE: Nextflow is not tested with Java 1.8.0_265 -- It's recommended the use of version 11 up to 18

N E X T F L O W  ~  version 22.04.1
Unexpected error [StackOverflowError]

$ NXF_VER=22.04.0 nextflow run nf-core/chipseq -profile test,docker
NOTE: Nextflow is not tested with Java 1.8.0_265 -- It's recommended the use of version 11 up to 18

N E X T F L O W  ~  version 22.04.0
Pulling nf-core/chipseq ...
 downloaded from https://github.com/nf-core/chipseq.git
Launching `https://github.com/nf-core/chipseq` [peaceful_darwin] DSL2 - revision: 6924b66942 [master]
Downloading plugin nf-tower@1.4.0
io/seqera/tower/plugin/TowerPlugin has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

**Comment by JoseEspinosa** [comment]
Actually, seems to work with the same command with `NXF_VER=22.04.0` i.e. setting `-dsl1` instead of letting the code guessing.

**Comment by ewels** [comment]
Figures, sounds like the error is in the new regex that detects the DSL flavour and that only runs if the `-dsl1` flag isn't set / config isn't set.

**Comment by JoseEspinosa** [comment]
Yeap, that's what I thought 👍

**Comment by drpatelh** [comment]
"You got 99 problems, son and the patch release ain't but one!" 🕺🏽

**Comment by scottgigante-immunai** [comment]
An additional nuance: I had exported `NXF_DEFAULT_DSL=1` and still got this error. So `-dsl1` works but the env var equivalent does not.

**Comment by pditommaso** [comment]
Solved by a05fcbea. Please upgrade to 22.04.2
